src/App.tsx

```diff
@@ -1,25 +1,42 @@
-import { useState } from "react";
-import "./App.css";
-import SortBy from "./components/SortBy";
-import UserSelctr from "./components/UserSelector";
-import ToDoLst from "./components/ToDoList";
+import { useState, useEffect } from 'react';
+import './App.css';
+import SortBy from './components/SortBy';
+import UserSelctr from './components/UserSelector';
+import ToDoLst from './components/ToDoList';
+import getApiData from './api';
 
 function App() {
-  const [selectedUser, setSelectedUser] = useState("1");
-  const [sortedBy, setSortedBy] = useState("default");
+  const [selectedUser, setSelectedUser] = useState('1');
+  const [sortedBy, setSortedBy] = useState('default');
+  const [usersData, setUsersData] = useState([]);
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    const fetchToDos = async () => {
+      setIsLoading(true);
+      let result = await getApiData();
+      setUsersData(result);
+      setIsLoading(false);
+    };
+    fetchToDos();
```

Comment:
Please use AbortController here. It will help to cancel the previous requests
https://developer.mozilla.org/en-US/docs/Web/API/AbortController

```suggestion
    const abortController = new AbortController();
  
    const fetchToDos = async () => {
      setIsLoading(true);
      let result = await getApiData(abortController.signal);
      setUsersData(result);
      setIsLoading(false);
    };
    fetchToDos();
    
    return () => {
      abortController.abort()
    }
```